sdk/samples/iot/hub/esp8266nodemcu/azure_iot_hub_telemetry.ino

```diff
@@ -281,4 +281,6 @@ void loop()
 
     next_telemetry_send_time_ms = millis() + TELEMETRY_FREQUENCY_MILLISECS;
   }
+  
+  mqtt_client.loop();
```

Comment:
This indeed seems to be missing from our code. We will add that. Thanks.

Reply:
Closing this PR as the main fix (lack of .loop() calls) was introduced by https://github.com/Azure/azure-sdk-for-c/pull/899